src/components/sidebar/sidebar.component.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import Icon from '../icon/icon';
+import Modal from '../modal/modal';
+import SidebarStyle from './sidebar.style';
+import './sidebar.scss';
+
+class Sidebar extends Modal {
+  /** Returns classes for the component. */
+  get mainClasses() {
+    return classNames(
+      'carbon-sidebar',
+      this.props.className
+    );
+  }
+
+  /** Returns the markup for the close icon. */
+  get closeButton() {
+    if (this.props.onCancel) {
+      return (
+        <span className='carbon-sidebar__close'>
+          <Icon
+            className='carbon-sidebar__close-icon'
+            data-element='close'
+            onClick={ this.props.onCancel }
+            type='close'
+          />
+        </span>
+      );
+    }
+    return null;
+  }
+
+  componentTags(props) {
+    return {
+      'data-component': 'sidebar',
+      'data-element': props['data-element'],
+      'data-role': props['data-role']
+    };
+  }
+
+  /** Returns the computed HTML for the sidebar. */
+  get modalHTML() {
+    return (
+      <SidebarStyle
+        position={ this.props.position }
+        size={ this.props.size }
+        { ...this.componentTags(this.props) }
+      >
+        { this.closeButton }
+        { this.props.children }
+      </SidebarStyle>
+    );
+  }
+
+  get transitionName() {
+    return `sidebar--${this.props.position}`;
```

Comment:
could that be moved to styled-components?

Reply:
not for now, it based on inheritance

Reply:
please leave a comment for it to be refactored or create a task for that